lib/api_catalogue.rb

```diff
@@ -10,6 +14,38 @@ def self.from_csv(catalogue_csv:, organisation_csv:)
     new(apis: apis, organisations: organisations)
   end
 
+  def self.from_url(url)
+    v1_alpha_fetcher = V1AlphaFetcher.new
+    FetcherService.new([v1_alpha_fetcher]).fetch(url)
+  rescue ClientError
+    ApiCatalogue.new(apis: [], organisations: [])
+  end
+
+  def self.from_urls(registry_entries_csv)
+    url_catalogues = []
+    registry_entries = CsvSource.load(registry_entries_csv) { |attributes| RegistryEntry.new(attributes) }
+    registry_entries.each do |entry|
+      catalogue = from_url(entry.url)
+      url_catalogues << catalogue
+    end
+    merge(url_catalogues)
+  end
+
+  def self.merge(api_catalogues)
+    all_apis = []
+    all_orgs = []
+    api_catalogues.each do |catalogue|
+      next unless catalogue.present?
```

Comment:
in what cases are these not present? If it's when it's empty, it's probably safe to continue through the loop, as they'll get ignored anyway